wallet-new/src/Cardano/Wallet/Kernel/Wallets.hs

```diff
@@ -384,7 +384,11 @@ updatePassword pw hdRootId oldPassword newPassword = do
                                -- was set.
                                -- Fix this properly as part of [CBR-404].
                                lastUpdateNow <- InDb <$> getCurrentTimestamp
-                               let hasSpendingPassword = HD.HasSpendingPassword lastUpdateNow
+
+                               let hasSpendingPassword = if (newPassword == emptyPassphrase)
+                                                         then HD.NoSpendingPassword
+                                                         else HD.HasSpendingPassword lastUpdateNow
```

Comment:
This is the way to go. However, also let's add a comment on _why_ we do this, as it might not be totally obvious to the reader that actually the frontend does allow the removal of the spending password after the wallet has been created (I, for one, got to know it just today)